core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -0,0 +1,720 @@
+    pub(crate) fn new(
+        consume_work_senders: Vec<Sender<ConsumeWork<Tx>>>,
+        finished_consume_work_receiver: Receiver<FinishedConsumeWork<Tx>>,
+        config: GreedySchedulerConfig,
+    ) -> Self {
+        let num_threads = consume_work_senders.len();
+        Self {
+            in_flight_tracker: InFlightTracker::new(num_threads),
+            account_locks: ThreadAwareAccountLocks::new(num_threads),
+            consume_work_senders,
+            finished_consume_work_receiver,
+            working_account_set: ReadWriteAccountSet::default(),
+            unschedulables: Vec::with_capacity(config.max_scanned_transactions_per_scheduling_pass),
+            config,
+        }
+    }
+}
+
+impl<Tx: TransactionWithMeta> Scheduler<Tx> for GreedyScheduler<Tx> {
+    fn schedule<S: StateContainer<Tx>>(
+        &mut self,
+        container: &mut S,
+        _pre_graph_filter: impl Fn(&[&Tx], &mut [bool]),
+        pre_lock_filter: impl Fn(&Tx) -> bool,
+    ) -> Result<SchedulingSummary, SchedulerError> {
+        let num_threads = self.consume_work_senders.len();
+        let max_cu_per_thread = self.config.max_scheduled_cus / num_threads as u64;
+
+        let mut schedulable_threads = ThreadSet::any(num_threads);
+        for thread_id in 0..num_threads {
+            if self.in_flight_tracker.cus_in_flight_per_thread()[thread_id] >= max_cu_per_thread {
+                schedulable_threads.remove(thread_id);
+            }
+        }
+        if schedulable_threads.is_empty() {
+            return Ok(SchedulingSummary {
+                num_scheduled: 0,
+                num_unschedulable: 0,
+                num_filtered_out: 0,
+                filter_time_us: 0,
+            });
+        }
+
+        // Track metrics on filter.
+        let num_filtered_out: usize = 0;
+        let total_filter_time_us: u64 = 0;
+        let mut num_scanned: usize = 0;
+        let mut num_scheduled: usize = 0;
+        let mut num_sent: usize = 0;
+        let mut num_unschedulable: usize = 0;
+
+        let mut batches = Batches::new(num_threads, self.config.target_transactions_per_batch);
+        while num_scanned < self.config.max_scanned_transactions_per_scheduling_pass
+            && !schedulable_threads.is_empty()
+            && !container.is_empty()
+        {
+            let Some(id) = container.pop() else {
+                unreachable!("container is not empty")
+            };
+
+            num_scanned += 1;
+
+            // Should always be in the container, during initial testing phase panic.
+            // Later, we can replace with a continue in case this does happen.
+            let Some(transaction_state) = container.get_mut_transaction_state(id.id) else {
+                panic!("transaction state must exist")
+            };
+
+            // If there is a conflict with any of the transactions in the current batches,
+            // we should immediately send out the batches, so this transaction may be scheduled.
```

Comment:
This comment was a little confusing. Can you elaborate that we are sending batches immediately because we want to start processing any blocking transactions immediately so that we can get them completed asap and unblock this high-priority transaction?

Reply:
Batches cannot have conflicting transactions (until SIMD-0083 activation). So we track which accounts are used in the batches we're currently building. If there's a conflict with those batches, we send the batches out to the workers now. That way the transaction that was conflicting with one of the batches could be inserted into a new batch.

Reply:
Yeah, I think we're on the same page here. Was just asking you to improve the code comment with my suggested elaboration (add yours too while you're at it 😉 )